apiv2/internal/server/os_update_run.go

```diff
@@ -5,31 +5,132 @@ package server
 
 import (
 	"context"
+	"time"
 
-	"google.golang.org/grpc/codes"
+	"google.golang.org/protobuf/types/known/timestamppb"
 
 	computev1 "github.com/open-edge-platform/infra-core/apiv2/v2/internal/pbapi/resources/compute/v1"
+	statusv1 "github.com/open-edge-platform/infra-core/apiv2/v2/internal/pbapi/resources/status/v1"
 	restv1 "github.com/open-edge-platform/infra-core/apiv2/v2/internal/pbapi/services/v1"
+	inv_computev1 "github.com/open-edge-platform/infra-core/inventory/v2/pkg/api/compute/v1"
+	inventory "github.com/open-edge-platform/infra-core/inventory/v2/pkg/api/inventory/v1"
 	"github.com/open-edge-platform/infra-core/inventory/v2/pkg/errors"
+	"github.com/open-edge-platform/infra-core/inventory/v2/pkg/validator"
 )
 
-func (is *InventorygRPCServer) ListOSUpdateRun(_ context.Context, _ *restv1.ListOSUpdateRunRequest) (
+func ConvertFromInvOSUpdatePolicy(policy *inv_computev1.OSUpdatePolicyResource) *computev1.OSUpdatePolicy {
```

Comment:
Instead of redefining it here, can you reuse `fromInvOSUpdatePolicy` defined into `os_update_policy.go`?